lib/updatehub/updatehub.c

```diff
@@ -714,10 +725,19 @@ enum updatehub_response updatehub_probe(void)
 				UPDATEHUB_INCOMPATIBLE_HARDWARE;
 			goto cleanup;
 		}
+
+		sha256size = strlen(
+			metadata_any_boards.objects[1].objects.sha256sum);
+
+		if (sha256size != TC_SHA256_BLOCK_SIZE) {
+			LOG_ERR("SHA256 size is invalid");
+			ctx.code_status = UPDATEHUB_METADATA_ERROR;
+			goto cleanup;
+		}
+
 		memcpy(update_info.sha256sum_image,
 		       metadata_some_boards.objects[1].objects.sha256sum,
-		       strlen(metadata_some_boards.objects[1]
-			      .objects.sha256sum));
+		       TC_SHA256_BLOCK_SIZE);
```

Comment:
Again, the null isn't getting set here (is it elsewhere?)

Reply:
Now we use total hex string size on memcpy operations.